adapters/beachfront/beachfront.go

```diff
@@ -383,103 +470,56 @@ func fallBackDeviceType(request *openrtb.BidRequest) openrtb.DeviceType {
 	return openrtb.DeviceTypeMobileTablet
 }
 
-func getVideoRequests(request *openrtb.BidRequest) ([]beachfrontVideoRequest, []error) {
-	var bfReqs = make([]beachfrontVideoRequest, len(request.Imp))
-	var errs = make([]error, 0, len(request.Imp))
-	var failedRequestIndicies = make([]int, 0)
-
-	for i := 0; i < len(request.Imp); i++ {
-		beachfrontExt, err := getBeachfrontExtension(request.Imp[i])
-
-		if err != nil {
-			// Failed to extract the beachfrontExt, so this request is junk.
-			failedRequestIndicies = append(failedRequestIndicies, i)
-			errs = append(errs, err)
-			continue
-		}
-
-		appid, err := getAppId(beachfrontExt, openrtb_ext.BidTypeVideo)
+func prepVideoRequestExt(requestImp openrtb.Imp) (openrtb_ext.ExtImpBeachfront, error) {
+	beachfrontExt, err := getBeachfrontExtension(requestImp)
 
-		if err != nil {
-			// Failed to get an appid, so this request is junk.
-			failedRequestIndicies = append(failedRequestIndicies, i)
-			errs = append(errs, err)
-			continue
-		}
+	if err != nil {
+		return beachfrontExt, err
+	}
 
-		bfReqs[i].AppId = appid
+	beachfrontExt.AppId, err = getAppId(beachfrontExt, openrtb_ext.BidTypeVideo)
 
-		if beachfrontExt.VideoResponseType != "" {
-			bfReqs[i].VideoResponseType = beachfrontExt.VideoResponseType
-		} else {
-			bfReqs[i].VideoResponseType = "nurl"
-		}
+	if err != nil {
+		return beachfrontExt, err
+	}
 
-		bfReqs[i].Request = *request
-		var secure int8
+	if beachfrontExt.VideoResponseType != "adm" && beachfrontExt.VideoResponseType != "both" {
+		beachfrontExt.VideoResponseType = "nurl"
+	}
 
-		if bfReqs[i].Request.Site != nil && bfReqs[i].Request.Site.Domain == "" && bfReqs[i].Request.Site.Page != "" {
-			bfReqs[i].Request.Site.Domain = getDomain(bfReqs[i].Request.Site.Page)
+	return beachfrontExt, err
+}
 
-			secure = isSecure(bfReqs[i].Request.Site.Page)
-		}
+func prepVideoRequest(bfReq videoRequest) videoRequest {
+	if bfReq.Request.Site != nil && bfReq.Request.Site.Domain == "" && bfReq.Request.Site.Page != "" {
+		bfReq.Request.Site.Domain = getDomain(bfReq.Request.Site.Page)
+	}
 
-		if bfReqs[i].Request.App != nil && bfReqs[i].Request.App.Domain == "" && bfReqs[i].Request.App.Bundle != "" {
-			if bfReqs[i].Request.App.Bundle != "" {
-				var chunks = strings.Split(strings.Trim(bfReqs[i].Request.App.Bundle, "_"), ".")
+	if bfReq.Request.App != nil && bfReq.Request.App.Domain == "" && bfReq.Request.App.Bundle != "" {
+		if bfReq.Request.App.Bundle != "" {
```

Comment:
Looks like this condition is redundant. It's present in the above if condition as well and so you can skip this next one